src/Planners/include/BipedalLocomotion/Planners/Contact.h

```diff
@@ -55,6 +55,11 @@ struct Contact
      */
     std::string name {"Contact"};
 
+    /**
+     * Current state of contact
+     */
+    bool isActive{false};
```

Comment:
Nitpick: The `activationTime` and `deactivationTime` seem to suggest that this data structure is used to specify a Contact + segment in time in which it is active, while this attribute `isActive` seems to mean something else. If the `activationTime` and `deactivationTime` is not used by the ContactDetector, probably it could make sense to use a separate structure.

Reply:
The thought was to provide also the switching times along with the contact state that will be an output of an arbitrary impact detection algorithm. Additionally, the contact frame poses within the structure comes in handy, because this structure could directly become a part of the [`FloatingBaseEstimators::InternalState`](https://github.com/dic-iit/bipedal-locomotion-framework/blob/67b78ab1f6942df37077c038f8c09242d2a09f46/src/Estimators/include/BipedalLocomotion/FloatingBaseEstimators/FloatingBaseEstimatorIO.h#L31) deprecating the currently existing pose for only feet contacts.

However, those fields are not filled by the `SchmittTriggerDetector` now.

The user needs to be aware that only one of the activation time and deactivation time will carry meaningful value depending on the contact transition. This can be documented in the `ContactDetector`. 

What would be your opinion on this?

Reply:
I realized that the concept of ``Contact`` here is different. The ``Contact`` class was originally conceived to be used in planners. In the planner case, the relevant data are the location and the timing, since there is a prediction horizon. The ``isActive`` field would not make sense in this case. I guess that in this case, the necessity is different, since you may detect that a contact is active, but you can't tell for how long, so the ``deactivationTime`` is meaningless. 

Some how, in one case the focus is on the future, in the other on the past, but both cases are interested in the location. At this point, I see two alternatives:
- we write a ``ContactBase`` struct with only the location and the type, then we have a  ``PlannedContact`` and an ``EstimatedContact``.
- we do not add ``isActive``, but for the measured case we use a negative number for ``activationTime``, and to check if it is active, you look at the ``deactivationTime``. If it is identically 0, it is active, if it is lower than 0 is not active. This is to say that right now you are sure, that the contact is active, but you can not say anything at ``t + dt``.

Reply:
I think the first alternative is more simple and seems better. I would not be totally afraid to also have two completely different data structure if it turns out to be more convenient.

Reply:
I also agree with this:
> we write a ContactBase struct with only the location and the type, then we have a PlannedContact and an EstimatedContact.

Reply:
I like @S-Dafarra's suggestion. 
But it means we need to change all instances of `Contact` to `PlannedContact` in the planner. And the rest of the data structures in the Contact library also incline towards `Contacts` that are planned.  So they must also be renamed accordingly.

Alternatively, we could just minimize the effort by defining a completely different structure within the `ContactDetector` library. This means that the currently existing PR https://github.com/dic-iit/bipedal-locomotion-framework/pull/141 in refactoring the Contact library out of Planners can also be closed.

For planned contact, we will have `Planners::Contact`.
For estimated contact, we will have `Estimators::Contact`.

The only difference would be the `*Time` fields.

Reply:
> Alternatively, we could just minimize the effort by defining a completely different structure within the ContactDetector library. This means that the currently existing PR #141 in refactoring the Contact library out of Planners can also be closed.

Why? It can still be helpful in the future :wink:

Reply:
> > Alternatively, we could just minimize the effort by defining a completely different structure within the ContactDetector library. This means that the currently existing PR #141 in refactoring the Contact library out of Planners can also be closed.
> 
> Why? It can still be helpful in the future

Ah yes, with a little thought, indeed it eases on the compilation.

Reply:
> Alternatively, we could just minimize the effort by defining a completely different structure within the ContactDetector library. This means that the currently existing PR #141 in refactoring the Contact library out of Planners can also be closed without merging.
>
>For planned contact, we will have Planners::Contact.
>For estimated contact, we will have Estimators::Contact.

What happens in case you do 
```c++
using namespace BipedalLocomotion::Planners;
using namespace BipedalLocomotion::Estimators;
```
?

Reply:
So as an eventual migration, I will create a new `Contacts` library independent from `Planners::Contact` library and create the `ContactBase` and `EstimatedContact` structs to be introduced within this PR.

We could then move the `PlannedContacts`  and related data structures in consequent PRs?

Reply:
It is also true that ``using namespace`` is considered a bad practice (https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice), and I like the symmetry, but it is also true that given our love for long names, we are basically forcing the user to do so.

Reply:
> So as an eventual migration, I will create a new `Contacts` library independent from `Planners::Contact` library and create the `ContactBase` and `EstimatedContact` structs to be introduced within this PR.
> 
> We could then move the `PlannedContacts` and related data structures in consequent PRs?
> 
> So that it just doesn't crash the other ongoing works on Python bindings?

Sounds good to me

Reply:
> We could then move the PlannedContacts and related data structures in consequent PRs?

Yep, if you do not have time I can handle the migration on the planner side

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/142/commits/5bd5f07fd11fe4cbef191e78a385bcf23e3eac9c

Reply:
> But it means we need to change all instances of Contact to PlannedContact in the planner. 

Not directly related to this, but especially for non-API stable libraries, and also for that one if we want to change the name of a class while keeping the deprecated one via a `typedef`, we should not be afraid of renaming a class just because it is used. Typically C++ IDEs provide tools to do that renaming automatically, and I think that such tooling is quite useful in practice.

Reply:
>  Typically C++ IDEs provide tools to do that renaming automatically, and I think that such tooling is quite useful in practice.

In KDevelop, there seems to be the option `Code -> RenameDeclaration`.